cmds/script.c

```diff
@@ -197,8 +197,10 @@ int command_script(int argc, char *argv[], struct opts *opts)
 
 	/* dtor for script support */
 	script_uftrace_end();
-out:
+
 	script_finish();
+out:
+	script_finish_filter();
```

Comment:
The `script_finish_filter()` is an API only for scripting engines (python, luafit) so I don't want to use it from outside.  I'd rather recommend skip calling `script_finish()` in case of failure.

Reply:
`script_finish()` is called to avoid memory leak previously. Otherwise, we better pass a flag to `script_finish` then avoid calling `script_finish_for_python`.

Reply:
Then it'd be a responsibility of `script_init()` to release all the memory in case of failure.

Reply:
Hmm.. may be right. I will see if it's okay.

Reply:
I updated it as you suggested. Thanks.